File: INW/Judge.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;


public class Judge {

	static int mode;
	static int[][] map = new int[1000][1000];
	static Scanner s = new Scanner(System.in);


	static void generateMap(int h, int w) {
		for (int i=0; i<h; ++i) for (int j=0; j<w; ++j) {
			map[i][j] = s.nextInt();
		}
	}

	static void singleCase(INW sol, int nr) {
		int h = s.nextInt();
		int w = s.nextInt();
		generateMap(h, w);

		sol.init(h, w, map);
		int res1 = sol.getInvasionDuration();
		int res2 = sol.getFinalArea();
		System.out.println("#" + nr + ": " + res1 + " " + res2);
	}

	static public void run(INW solution) {
		mode = s.nextInt();
		int tc = s.nextInt();

		for (int i=0; i<tc; ++i)
			singleCase(solution, i+1);
	}
}

File: INW/judge.cpp
#include "judge.h"
#include <iostream>
#include <fstream>
#include <chrono>
using namespace std;

namespace {

class MyTimer {
public:
    MyTimer(const string& _name) : name(_name) {
        start = chrono::high_resolution_clock::now();
    }

    MyTimer(char _name) {
        name = _name;
        start = chrono::high_resolution_clock::now();
    }

    ~MyTimer() {
        cout << name << " " << chrono::duration_cast<chrono::milliseconds>(chrono::high_resolution_clock::now() - start).count() << " ms" << endl;
    }
private:
    string name;
    chrono::time_point<chrono::high_resolution_clock> start;
};

//string TestCases = "b";
string TestCases = "abc";
string TestCasePrefix = "res/INW0";
string TestCaseSuffixIn = ".in";
string TestCaseSuffixOut = ".out";

int mode;
int map[1000][1000];

void generateMap(int h, int w, ifstream& ins) {
    for (int i=0; i<h; ++i) for (int j=0; j<w; ++j) {
        ins >> map[i][j];
    }
}

void singleCase(IINW *sol, int nr, ifstream& ins, ifstream& outs) {
    int h; ins >> h;
    int w; ins >> w;
    generateMap(h, w,ins);

    if (nr == 4) {
        int aaa = 0;
    }

    sol->init(h, w, map);
    int res1 = sol->getInvasionDuration();
    int res2 = sol->getFinalArea();

    string sss;
    int duration;
    int area;
    outs >> sss >> duration >> area;

    if (res1 != duration || res2 != area) {
        cout << "#" << nr << ": " << res1 << "/" << duration << " " << res2 << "/" << area << endl;;
    } else {
        cout << "#" << nr << " OK" << endl;;
    }
}

} // namespace

void Judge::run(IINW *solution) {

    for (auto cc : TestCases) {
        ifstream ins(TestCasePrefix + cc + TestCaseSuffixIn);
        ifstream outs(TestCasePrefix + cc + TestCaseSuffixOut);

        MyTimer timer(cc);

        ins >> mode;
        int tc; ins >> tc;

        for (int i=0; i<tc; ++i)
            singleCase(solution, i+1,ins,outs);
    }
}

